taichi/codegen/spirv/spirv_ir_builder.cpp

```diff
@@ -328,6 +328,27 @@ SType IRBuilder::get_primitive_type(const DataType &dt) const {
   }
 }
 
+SType IRBuilder::from_taichi_type(const DataType &dt, bool has_buffer_ptr) {
+  if (dt->is<PrimitiveType>()) {
+    return get_primitive_type(dt);
+  } else if (dt->is<PointerType>()) {
+    if (has_buffer_ptr) {
+      return t_uint64_;
+    } else {
+      return t_uint32_;
```

Comment:
ditto, what's this?

Reply:
Same as above. When has_buffer_ptr is false, we need a placeholder argument. It was the element type of the pointer before. In this PR I changed it to u32 for simplicity.